tool/tsh/common/logger.go

```diff
@@ -0,0 +1,84 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"log/slog"
+	"os"
+	"strconv"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	debugEnvVar = teleport.VerboseLogsEnvVar // "TELEPORT_DEBUG"
+	osLogEnvVar = "TELEPORT_OS_LOG"
+)
+
+// initLogger initializes the logger.
+//
+// It is called twice, first soon after launching tsh before argv is parsed and then again after
+// kingpin parses argv. This makes it possible to debug early startup functionality, particularly
+// command aliases.
+func initLogger(cf *CLIConf, opts debugOpts) error {
+	cf.OSLog = opts.osLog
+	cf.Debug = opts.debug || opts.osLog
+
+	loggerOpts := getPlatformInitLoggerOpts(cf)
+
+	level := slog.LevelWarn
+	if cf.Debug {
+		level = slog.LevelDebug
+	}
+
+	return trace.Wrap(utils.InitLogger(utils.LoggingForCLI, level, loggerOpts...))
+}
+
+type debugOpts struct {
+	debug bool
+	osLog bool
+}
+
+// parseDebugOptsFromEnv calculates debug opts taking into account only env vars.
+func parseDebugOptsFromEnv() debugOpts {
+	var opts debugOpts
+	opts.debug, _ = strconv.ParseBool(os.Getenv(debugEnvVar))
+	opts.osLog, _ = strconv.ParseBool(os.Getenv(osLogEnvVar))
```

Comment:
We might want to log those, I guess using `slog.Default()`?

Reply:
idk if it's worth it. 😩 If someone wants to log to os_log, I'd rather avoid _always_ outputting some logs to stdout despite `--os-log` being used.

Reply:
Fair enough; I think we can reframe `--os-log` to mean "we will _try_ to use OS logging" rather than "we will _never_ log anything to stderr", but ultimately quietly ignoring this error isn't critical.